src/typescript/azure-openapi-validator/rules/UniqueClientParameterName.ts

```diff
@@ -0,0 +1,36 @@
+import { nodes } from "jsonpath"
+import { MergeStates, OpenApiTypes, rules } from "./../rule"
+
+export const UniqueClientParameterName: string = "UniqueClientParameterName"
+
+rules.push({
+  id: "R4026",
+  name: UniqueClientParameterName,
+  severity: "error",
+  category: "SDKViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.arm | OpenApiTypes.dataplane,
+  appliesTo_JsonQuery: "$",
+  *run(doc, node, path) {
+    const clientParameterName = new Set<string>()
+    for (const it of nodes(node, '$..["parameters"]')) {
```

Comment:
Don't use "$.." , use "$.paths.*.*.parameters" & "$.paths.*.parameters" instead .  Because the example or other places may contains the "parameters" as well.
Also forget telling you only consider the parameter without  '"x-ms-parameter-location": "method"' , see details here: https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameter-location